src/App.tsx

```diff
@@ -17,18 +45,30 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                option={typeOfTodos}
+                setTypeOfTodos={setTypeOfTodos}
+                filter={filter}
+                setFilter={setFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {
+                isLoading && <Loader />
+              }
+              <TodoList
+                todos={filteredTodos}
+                onButtonClick={setTodoToShow}
```

Comment:
don't pass function for updating state in child component, create wrapper for it, fix in all places